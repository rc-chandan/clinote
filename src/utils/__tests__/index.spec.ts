import { checkOSSupport } from '../';

describe('Supported OS types should be linux, macos and windows only', () => {
  
  test('app should support Darwin', () => {
    expect(checkOSSupport('Darwin')).toBe(true);
  });

  test('app should support Linux', () => {
    expect(checkOSSupport('Linux')).toBe(true);
  });

  test('app should support Windows_NT', () => {
    expect(checkOSSupport('Windows_NT')).toBe(true);
  });

  test('clinote should throw error for unknown os type', () => {
    try {
      expect(checkOSSupport('windows_mobile')).toThrow();
    } catch (error) {
      expect(error).toMatchSnapshot();
    }
  });

});
