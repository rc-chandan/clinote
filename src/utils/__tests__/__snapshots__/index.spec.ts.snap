// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Supported OS types should be linux, macos and windows only clinote should throw error for unknown os type 1`] = `[Error: windows_mobile is not supported by clinote,supported OS list: Linux,Darwin,Windows_NT]`;
